### PR TITLE
Fixup regex for issue submitter, Dont allow dupe app submitted issues. More

### PR DESCRIPTION
**Pull request by miigotu**
Duplicate submitted issues are now comments to the original instead of
duplicating. ***DO NOT CHANGE APP SUBMITTED ISSUE TITLES FOR NOW**
Maybe later we can move to parsing issue comments to match errors and
make it ok to change APP SUBMITTED issue titles, but currently I have it using
the issue title to search for originals.

Fix some unit tests not running some or all of their tests when ran standalone.

See how it just adds a comment instead of a  new issue in SiCKRAGETV/sickrage-issues/issues/1392

Also, the gists it pasts are just the error it finds, not the entire log, which was the original writer's intent.

### CODE REVIEW

File: sickbeard/logger.py [hunk 1]
```diff
@@ -151,6 +151,7 @@ def log_error_and_exit(self, error_msg, *args, **kwargs):
 
     def submit_errors(self):
         if not (sickbeard.GIT_USERNAME and sickbeard.GIT_PASSWORD and len(classes.ErrorViewer.errors) > 0):
+            print u'Please set your GitHub username and password in the config, unable to submit issue ticket to GitHub!'
```
Comment (abeloin) [inline sickbeard/logger.py]: @miigotu This should be a logger.log instead of a print

**Comment by miigotu** [comment]
@fernandog what do you think about this?

**Comment by miigotu** [comment]
oops, fixing.

**Comment by miigotu** [comment]
@abeloin @fernandog Should be all gravy now.

**Comment by fernandog** [comment]
@miigotu thanks! Log file fixed and None title in some cases too? @tagforce find out that in some cases the title is None or Empty.

**Comment by fernandog** [comment]
@miigotu the problem is log rotation. That's why we read all log files. If the error is in sickrage.log.1 after a log rotation , the submit wouldn't work when user press "submit errors"

**Comment by fernandog** [comment]
@miigotu do we need to update this regex also?

https://github.com/SiCKRAGETV/SickRage/blob/develop/sickbeard/webserve.py#L5039

**Comment by miigotu** [comment]
@fernandog That regex is working for every line in my log, however it could be improved for performance by changing it to this: https://regex101.com/r/mE8wS5/1
Notice the "X Match - Y Steps" at the top, now look at the original:
https://regex101.com/r/hK2tX4/1

This regex is used for each line in the log, so a 17 step savings in this particular regex is a savings of approximately 30% processing power. Less instructions = More performance.

It could be improved further by not capturing the date and time as well, which saves two steps.
Either way, if changing the regex then the capture group numbers would need changed to correctly represent the match we are looking for in:

https://github.com/SiCKRAGETV/SickRage/blob/develop/sickbeard/webserve.py#L4981
https://github.com/SiCKRAGETV/SickRage/blob/develop/sickbeard/webserve.py#L4982

It works as it is, but performance can be improved. Shall we leave it alone or who shall PR?

**Comment by fernandog** [comment]
@abeloin what do you think?

**Comment by fernandog** [comment]
https://github.com/SiCKRAGETV/sickrage-issues/issues/1282

**Pull request merged**